### PR TITLE
Feature in-tree external components

### PR DESCRIPTION
**Pull request by untereiner**
A proposal to solve issue #2526

### CODE REVIEW

**Comment by untereiner** [comment]
@TotoGaz Could you take a look please ?

**Review by TotoGaz — APPROVED** [review]

File: src/cmake/GeosxOptions.cmake [hunk 1]
```diff
@@ -127,7 +127,7 @@ blt_append_custom_compiler_flag( FLAGS_VAR CMAKE_CXX_FLAGS
                                )
 
 blt_append_custom_compiler_flag( FLAGS_VAR CMAKE_CXX_FLAGS_DEBUG
-                                 GNU "-Wno-unused-parameter -Wno-unused-variable"
+                                 GNU "-Wno-unused-parameter -Wno-unused-variable -Wno-dangling-reference"
```
Comment (rrsettgast) [inline src/cmake/GeosxOptions.cmake:130]: What was this required for? i.e. where is the dangling reference that this avoids?

**Review by rrsettgast — APPROVED** [review]
@untereiner Thanks for picking this up. I don't think anyone has used this since we ditched PAMELA and moved PVTPackage as a submodule inside the `src` directory. Have you confirmed that this works with an external package that is called at runtime?

**Comment by untereiner** [comment]
@rrsettgast Yes I confirm. I am using RESQML inputs/outputs with a kind of [plugin](https://github.com/geosiris-technologies/geos-resqml/)

**Comment by rrsettgast** [comment]
> @rrsettgast Yes I confirm. I am using RESQML inputs/outputs with a kind of [plugin](https://github.com/geosiris-technologies/geos-resqml/)

Great. Thanks again.

**Pull request merged**